### ISSUE TITLE
IPN delay from PayPal causing Order not found error

### ISSUE THREAD

**Issue by andrelopez** [enupal/paypal#7]
IPN is not real-time notification
Add a payment status to the order 

- Pending
- Succeeded

Instead of returning Order Not fund when the user is redirected to the site, create a pending Order.

**Comment by cole007** [comment]
Having issues with this too on client site. Oddly the Paypal account we were testing with on smaller transactions worked fine but with client's Paypal Merchant ID the callback is timing out so the order is not completing at our end.

Perhaps one solution would be to make sleep a configurable value? Eg default to 10 but let this be extended via an ENV variable or similar

**Comment by andrelopez** [comment]
@cole007 I agree, I am looking to add this config setting soon. I'll let you know once is released

**Comment by cole007** [comment]
Great thanks @andrelopez - in the meantime do you have any thoughts why one Merchant ID works fine but another times out?

Even if I hard-code the sleep variable in the `actionCompletePayment` controller to 30 seconds the order is not completing and an "Order does not exists" error is thrown. 

The payment is definitely taken but "the handshake" with the CMS is not completed.

**Comment by andrelopez** [comment]
@cole007 I will try to add the Pending status and change the status to Completed once the webhook is hit.

**Comment by cole007** [comment]
Hmmm, I suspect it is not a time out issue. Looking at the database it looks like the Order records is not being created so the "Order does not exists" error is because there is literally no order in the DB.

Any thoughts why one Merchant ID is generating Orders fine but another is not?

**Comment by andrelopez** [comment]
@cole007 To troubleshoot:

- Please check that the website can be reached publicly. (Try to access to the site from a incognito browser window) chances are that your site is not enabled.
- Please ask your client review IPN history:

> Review the IPN History for your account. Under History, click IPN History. This will give you the status of each IPN message PayPal generated. Messages must be marked as Sent for PayPal to mark them as having been delivered. Your listener must respond to every IPN message it gets, whether you act on it or not:
> If your IPN listener doesn't respond, PayPal assumes the IPN was never received and re-sends it. PayPal IPN will continue to re-send the message periodically until your listener responds, although the interval between retries increases with each attempt. An IPN will be re-sent for up to four days.
> If the IPNs are marked as Sent, this means PayPal delivered them to your IPN handler (the URL of which will be available too) and got an HTTP 200 OK response from your IPN handler.
> If PayPal doesn't receive an HTTP 200 response, for whatever reason (SSL handshake validation, network timeout, script error 500, server error 503, and so on), then PayPal will retry delivery of the IPN message up to an additional 15 times.

Please let me know how it goes.
